bumble/device.py

```diff
@@ -1717,6 +1731,9 @@ def on_connection_failure(self, connection_handle, error_code):
             'hci',
             HCI_Constant.error_name(error_code)
         )
+        error.transport = transport
+        error.connection_handle = connection_handle # FIXME: Connection handle sounds to be a dummy value here
+        error.peer_address = peer_address
```

Comment:
I suggest making `peer_address` a constructor parameter for `ConnectionError` (with a default to None)